test/functional/core/path_spec.lua

```diff
@@ -0,0 +1,61 @@
+local helpers = require('test.functional.helpers')(after_each)
+local clear = helpers.clear
+local eq = helpers.eq
+local eval = helpers.eval
+local get_pathsep = helpers.get_pathsep
+local command = helpers.command
+
+describe("'%:p' expanding", function()
+  local pathsep
+  local targetdir
+  local expected_path
+
+  local function get_full_path()
+    return eval('expand("%:p")')
+  end
+
+  local function join_path(...)
+    return table.concat({...}, pathsep)
+  end
+
+  before_each(function()
+    clear()
+    pathsep = get_pathsep()
+    targetdir = join_path('test', 'functional', 'fixtures')
+    clear(join_path(targetdir, 'tty-test.c'))
```

Comment:
`clear()` is somewhat expensive, specially to do it +2 times per test. So I removed the first case and move the other case to `setup()` (which only runs once per `describe()` block)